File: backend/src/main/java/com/shiptrack/backend/service/DriverService.java
package com.shiptrack.backend.service;

import com.shiptrack.backend.entity.Driver;

import java.util.List;

public interface DriverService {

    List<Driver> getAllDrivers();

    Driver getDriverById(Long id);

    Driver createDriver(Driver driver);

    Driver updateDriver(Long id, Driver driver);

    void deleteDriver(Long id);

}

File: backend/src/main/java/com/shiptrack/backend/service/driverService.js
import api from "./api";

export const getAllDrivers = () =>
    api.get("/admin/drivers");

export const getDriverById = (id) =>
    api.get(`/admin/drivers/${id}`);

export const createDriver = (driver) =>
    api.post("/admin/drivers", driver);

export const updateDriver = (id, driver) =>
    api.put(`/admin/drivers/${id}`, driver);

export const deleteDriver = (id) =>
    api.delete(`/admin/drivers/${id}`);